### DISCUSSION TITLE
Support for WSL in Windows

### DISCUSSION THREAD

**Discussion by anokun7** [InterDigitalInc/AdvantEDGE#168] [Q&A] [answered]
Can meepctl & AdvantEdge be deployed using WSL on Windows? 

I was able to install the meepctl tool and it seems to work, but when I run `meepctl config ip`, regardless of what ip address I provide, I get the error:
```
Using repo config file: /mnt/c/Users/project/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/.meepctl.yaml
Invalid IP: Not a local IP address [127.0.1.1]

Get/Set node IP address in the meepctl config file

Usage:
  meepctl config ip [IP] [flags]

Examples:
meepctl config ip 1.2.3.4

Flags:
  -h, --help   help for ip

Global Flags:
  -t, --time      Display timing information
  -v, --verbose   Display debug information
========================================
node.ip : 70.0.0.10
========================================
```

What ip address should be provided & should I expect this to work on WSL for Windows?

**Comment by roymx** [comment, marked as answer]
Hello anokun7,

Thanks for reaching out.

We did not test AdvantEDGE on Windows WSL, so I cannot definitively confirm if it will run or not.
The typical environments we use/test are either local Linux machine, cluster of multiple Linux machines and Linux VMs.

The error you are getting with configuring meepctl IP address is related to the fact that it's not recognized as _a valid local IP address_.
Which means meepctl did not find this IP on any of the interface of the WSL instance.
The IP address provided to meepctl (1) must not be local loopback (127.0.0.1), (2) must be IPv4 format and (3) must be valid local IP.

In a shell, do the following command `ifconfig` and use the IP address assigned to the network interface of your WSL instance (on my Ubuntu system, the interface is named `eno1` - it may have a different name in the WSL instance) - then the command should be accepted.

We think that what you are trying to achieve is cool, let us know if you need more help or if you succeed.

**Comment by anokun7** [comment]
Hi @roymx - Yes, that worked. I'm unblocked for now. Thank you.
